src/game_actor.cpp

```diff
@@ -293,10 +298,10 @@ int Game_Actor::GetSp() const {
 }
 
 int Game_Actor::GetBaseMaxHp(bool mod) const {
-	// Looks like RPG_RT only applies Class changes (changed_class == true)
+	// Looks like RPG_RT only applies Class changes (class_id > 0 - 20kdc)
 	// when the class was changed by the ChangeClass event, otherwise it uses
 	// the normal actor attributes.
-	int n = GetData().changed_class && GetData().class_id > 0
+	int n = GetData().class_id > 0
```

Comment:
So you want to tell us that the bug is liblcf which copies the class_id from Actor to SaveActor?